underscore.js

```diff
@@ -136,15 +136,35 @@
     };
   };
 
+  // Helper to determine if index is within the bounds of an array.
+  var isIndex = function(index, length) {
```

Comment:
Not necessarily, but there's a bit of duplication between `isArrayLike` and `isIterationCall` that'll get worse once I propose only considering ints and not floats as proper index.